### ISSUE TITLE
Still getting dot notation is deprecated warning

### ISSUE THREAD

**Issue by ltamaster** [grails-plugins/grails-spring-security-core#738]
### Expected Behavior

`conf.filterChain` shoundn't return a NavigableMap object

https://github.com/grails/grails-spring-security-core/blob/5.0.x/plugin/src/main/groovy/grails/plugin/springsecurity/ReflectionUtils.groovy#L212

![Screenshot from 2022-03-28 17-14-07](https://user-images.githubusercontent.com/6034968/160479440-d1200feb-3075-4a77-9f7e-76ad8b484e87.png)

### Actual Behaviour

this code returns a navigable map `grailsConfig.getProperty('grails.plugin.springsecurity', ConfigObject.class)` in `filterChain` field

https://github.com/grails/grails-spring-security-core/blob/5.0.x/plugin/src/main/groovy/grails/plugin/springsecurity/ReflectionUtils.groovy#L119

![Screenshot from 2022-03-28 17-11-05](https://user-images.githubusercontent.com/6034968/160479194-1db8df8e-7258-40d6-9157-d8413c816587.png)

### Steps To Reproduce

I reproduced it with this test:

```
void 'findFilterNames navigableMap'() {
		when:

		def chainMap = [
				[pattern: '/assets/**',      filters: 'none'],
				[pattern: '/**/js/**',       filters: 'none'],
				[pattern: '/**/css/**',      filters: 'none'],
				[pattern: '/**/images/**',   filters: 'none'],
				[pattern: '/**/favicon.ico', filters: 'none'],
				[pattern: '/**', 			 filters: 'JOINED_FILTERS']
		]

		grailsApplication.config.grails.plugin.springsecurity.filterChain.chainMap  = chainMap


		def config = ReflectionUtils.getSecurityConfig()
		def result = ReflectionUtils.findFilterChainNames(config)

		then:
		result != null
	}
```

### Environment Information

- OS: Ubuntu:
- java: 8 and 11

### Example Application

https://github.com/rundeck/rundeck

### Version

5.1.4

**Comment by puneetbehl** [comment]
Could you please verify if this is an issue with Grails 5.1.7?

**Comment by lynchie14** [comment]
Still happening on 5.1.7

```
2022-05-19 15:50:56.005  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Attempting to load [0] user defined plugins
2022-05-19 15:50:56.006  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [eventBus] with version [SNAPSHOT] loaded successfully
2022-05-19 15:50:56.006  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [restResponder] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.006  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [mail] with version [3.0.0] loaded successfully
2022-05-19 15:50:56.006  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [i18n] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.006  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [cascadeValidation] with version [3.0.1] loaded successfully
2022-05-19 15:50:56.006  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [core] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.007  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [databaseMigration] with version [4.1.1.SYSNET] loaded successfully
2022-05-19 15:50:56.007  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [codecs] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.008  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [urlMappings] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.008  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [dataSource] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.008  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [groovyPages] with version [5.1.0] loaded successfully
2022-05-19 15:50:56.008  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [controllers] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.008  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [mimeTypes] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.008  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [controllersAsync] with version [SNAPSHOT] loaded successfully
2022-05-19 15:50:56.008  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [domainClass] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.009  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [hibernate] with version [7.2.2] loaded successfully
2022-05-19 15:50:56.009  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [services] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.009  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [converters] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.009  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [interceptors] with version [5.1.7] loaded successfully
2022-05-19 15:50:56.009  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [springSecurityCore] with version [5.0.0] loaded successfully
2022-05-19 15:50:56.009  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [quartz] with version [2.0.13] loaded successfully
2022-05-19 15:50:56.009  INFO --- [  restartedMain] g.plugins.DefaultGrailsPluginManager     :  Grails plug-in [springSecurityRest] with version [3.0.1] loaded successfully

Configuring Spring Security Core ...
... finished configuring Spring Security Core


Configuring Spring Security REST 3.0.1...
... finished configuring Spring Security REST

2022-05-19 15:50:58.381  WARN --- [  restartedMain] org.grails.config.NavigableMap           :  Accessing config key '[filterNames]' through dot notation is deprecated, and it will be removed in a future release. Use 'config.getProperty(key, targetClass)' instead.
.....
2022-05-19 15:51:07.492  WARN --- [  restartedMain] org.grails.config.NavigableMap           :  Accessing config key '[filterNames]' through dot notation is deprecated, and it will be removed in a future release. Use 'config.getProperty(key, targetClass)' instead.
2022-05-19 15:51:07.493  WARN --- [  restartedMain] org.grails.config.NavigableMap           :  Accessing config key 'chainMap' through dot notation is deprecated, and it will be removed in a future release. Use 'config.getProperty(key, targetClass)' instead.
```

**Comment by puneetbehl** [comment]
This should be resolved with updating to Grails 5.1.8
